### DISCUSSION TITLE
when I code hexo g and TypeError: rule is required!

### DISCUSSION THREAD

**Discussion by BoiDontCry** [hexojs/hexo#5352] [Question / Answer] [answered]
I have previously uploaded articles on my blog. BUT idk what's wrong wiht it.When i tried to download some new plugin.Later, the blog encountered a 404 issue. After reinstalling several plugins and using 'hexo clean hexo g' again, I encountered the following error when using 'hexo g'.
![QQ图片20231112234312](https://github.com/hexojs/hexo/assets/128986918/30bdaf7c-988d-4792-b662-7e21ad00a19c)

**Comment by uiolee** [comment, marked as answer]
can't reproduce. Please provide more informations.

try to check your if `permalink` setting in `_config.yml`  and Front-matter of your posts are correct.

**Reply by BoiDontCry** [reply]
thanks to ur reply!!! now the problem has been resovled!!!
